risc0/build/src/lib.rs

```diff
@@ -426,8 +427,32 @@ pub(crate) fn cargo_command_internal(subcmd: &str, guest_info: &GuestInfo) -> Co
     cmd
 }
 
+fn get_rustc_version() -> Result<semver::Version> {
+    let output = Command::new("rustc")
+        .arg("--version")
+        .env("RUSTUP_TOOLCHAIN", "risc0")
+        .output()
+        .expect("failed to get risc0 toolchain's rustc version. Please ensure that your risc0 toolchain installed is correctly");
```

Comment:
If instead we could as `rzup` for the path to the Rust toolchain that would be better, maybe re-use the existing `rust_toolchain()` function in `risc0/build/src/lib.rs`